pkg/server/endpoints/registration/handler.go

```diff
@@ -545,6 +547,95 @@ func (h *Handler) ListAgents(ctx context.Context, listReq *registration.ListAgen
 	return &registration.ListAgentsResponse{Nodes: resp.Nodes}, nil
 }
 
+func (h *Handler) MintX509SVID(ctx context.Context, req *registration.MintX509SVIDRequest) (_ *registration.MintX509SVIDResponse, err error) {
+	counter := telemetry_registrationapi.StartMintX509SVIDCall(h.Metrics)
+	addCallerIDLabel(ctx, counter)
+	defer counter.Done(&err)
+
+	spiffeID, err := h.normalizeSPIFFEIDForMinting(req.SpiffeId)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(req.Csr) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "request missing CSR")
+	}
+
+	for _, dnsName := range req.DnsNames {
+		if err := validateDNS(dnsName); err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "%q is not a valid DNS name: %v", dnsName, err)
+		}
+	}
+
+	csr, err := x509.ParseCertificateRequest(req.Csr)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid CSR: %v", err)
+	}
+	if err := csr.CheckSignature(); err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid CSR: signature verify failed")
+	}
+
+	svid, err := h.ServerCA.SignX509SVID(ctx, ca.X509SVIDParams{
```

Comment:
We should check for error here. Any chance sensitive data can be returned?

Reply:
internal is probably ok. i don't think any sensitive information is returned from here.